pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -412,7 +414,8 @@ func (qjm *XController) PreemptQueueJobs(inspectAw *arbv1.AppWrapper) {
 			}
 
 			updateNewJob = newjob.DeepCopy()
-		} else {
+			generatedCondition = true
+		} else if newjob.Status.Running == 0 && newjob.Status.Succeeded == 0 && newjob.Status.State == arbv1.AppWrapperStateActive {
```

Comment:
I don't think this if is the right statement to have here. If this is true (i.e., Running = 0 & Succeeded = 0 & State = Active) then line 379 is true as well, so this if is useless. I think it should be `else if newjob.Status.State == arbv1.AppWrapperStateActive` and that is it. That means somehow the AppWrapper has the right number of running and succedeed pods equal to minAvailable, but it is not Active somehow. Then it should be requeued.

Reply:
Won't we land here when minAvailable is not set?

Reply:
That is actually handled in the function that returns preemtable AppWrappers (i.e., `GetQueueJobEligibleForPreemption`). If they don't have `minAvailable` then they are not returned so they wouldn't go inside here at all. I'm starting to think we may not even need this `else if` or `else` statements at all. What's inside this `if` was from the old code and may not apply anymore since we are handling scheduling failures above. I think we should remove it, what do you think?

Reply:
Pods can always fail to get scheduled we need to propagate such information to the users, so I think we need this loop.

Reply:
I don't see how that would be a problem. If they're not scheduled their `Running` and `Succeeded` counts would be 0 and there are other parts of the code that handle that logic. This `else if` is never going to be executed so why have it there anyway?

Reply:
if there are 0 pods running and minAvailable is not set then the previous if block will never be executed, this would mean that pods failed scheduling and else if will be triggered.

Reply:
Partially true. If `minAvailable` is not set then we would never come to this part of the code because line 504 would never pass and thus the AppWrapper is not a "preemtable" AppWrapper. As in line 326 `aw` would be `nil` and it wouldn't go into this `if` statement.